### PR TITLE
feat: add /client_config endpoint to view runtime client config

### PR DESCRIPTION
**Pull request by ppca**
add endpoint /client_config to dump runtime client config

# Context
https://pagodaplatform.atlassian.net/browse/ND-273  

# Testing and QA
tested by starting a localnet and tested endpoint on localnet
local_client_config

### CODE REVIEW

**Comment by wacban** [comment]
Please have a look at the failing tests in buildkite.
Can you make the json pretty (multiline)?

**Comment by ppca** [comment]
> Please have a look at the failing tests in buildkite. Can you make the json pretty (multiline)?

buildkite is good now :)
I created a ticket for making the json output nice: https://pagodaplatform.atlassian.net/browse/ND-282, we should make it easy to apply to all endpoints. Looks reasonable to be a separate task.

**Comment by wacban** [comment]
@ppca n00b question here: Do you know how to actually change the config to see that it's updated in the ui that you added? I mean changing it dynamically (what Nikolay is working on) rather than stopping neard, changing it in the config file and restarting neard.

File: core/chain-configs/src/client_config.rs [hunk 1]
```diff
@@ -70,7 +70,7 @@ impl GCConfig {
     }
 }
 
-#[derive(Clone, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
```
Comment (nikurt) [inline core/chain-configs/src/client_config.rs:73]: Due to another PR merging, please re-implement `Debug` and `Serialize` by implementing them for `MutableConfigValue`.

**Comment by nikurt** [comment]
> @ppca n00b question here: Do you know how to actually change the config to see that it's updated in the ui that you added? I mean changing it dynamically (what Nikolay is working on) rather than stopping neard, changing it in the config file and restarting neard.

```
echo '{"verbose_module": ""}' > ~/.near/log_config.json
killall -SIGHUP neard
```

Same for `config.json`. Change `expected_shutdown` and send `SIGHUP`.

**Comment by wacban** [comment]
I tried `killall -SIGHUP neard` but it complaned that dyn_config doesn't exist. I put the expected_shutdown in dyn_config and with sighup I was able to make it work (which is a fancy way of saying that I unintentionally killed neard by setting the shutdown height too low). This however is not part of client config and so it doesn't show up when testing this PR.

**Comment by wacban** [comment]
One thing bothers me with this approach: you're making the client config a part of the general use RPC which I believe node owners typically have open to anyone. I'm not sure if all node owners will want to make their configs public. Could we make this part of the debug page instead? Node owners that care abour privacy can disable that with enable_debug_rpc = false. It may simplify the implementation as well because you'd only be adding a subpage rather than a full RPC method though I may be wrong.

**Comment by nikurt** [comment]
> `SIGHUP` and `dyn_congig`

Before #8240 was merged, you could have `dyn_config.json` was checked, but if the file wasn't available

**Comment by wacban** [comment]
> After https://github.com/near/nearcore/pull/8240, dyn_config.json is no longer a thing. The only field that can be updated at runtime is expected_shutdown in config.json, but we can add more :)

hehe, ok, I'm behind, I'll rebase on top of master and test again ;)

**Comment by nikurt** [comment]
> privacy

None of the fields of `ClientConfig` are sensitive. `/client_config` is a handler on port 3030, which usually validators don't open. Some validators share their metrics and debug pages with us. Both `/client_config` and `/debug/client_config` would work.

Overall, I agree with @wacban, it's a good idea to be a bit caution, and make this a debug handler. Would be even better to add a link from `/debug` to the `ClientConfig` handler. Helps with discover-ability of this handler. Maybe we also should link `/status` and `/metrics` :thinking: but let's not go too far in this PR yet :)

**Pull request closed without merging**